File: Arrays/MoveZeroes/moveZeroes.cpp
#include<iostream>
using namespace std;

void moveZeroesToEnd(int arr[], int n){
    int count = 0; 
    for(int i = 0;i<n;i++){
    if(arr[i]!=0){
        swap(arr[i],arr[count]);
        count++;
    }
    }
}

int main()
{
    int arr[] = {1,2,3,0,0,0,0,0,0,0,0,0,0,5,6,0,0};
    int n = sizeof(arr)/sizeof(arr[0]);
        moveZeroesToEnd(arr,n);
    for(int i = 0;i<n; i++){
        cout<<arr[i]<<" ";
    }
    return 0;
}

File: Arrays/MoveZeroes/moveZeroes.java
package Arrays.MoveZeroes;

public class moveZeroes {


    public static void moveZeroesToEnd(Integer arr[]){
        Integer count = 0; 
        for(int i = 0; i<arr.length;i++){
            if(arr[i]!=0){
                Integer temp = arr[count];
                arr[count] = arr[i];
                arr[i] = temp; 
                count++;
            }
        }
    }

    public static void main(String args[]){
        Integer arr[] = {1,2,3,0,0,0,0,0,0,0,0,0,0,5,6,0,0};
        moveZeroesToEnd(arr);
        for(Integer x: arr){
            System.out.println(x);
        };
    }
    
}
